Add documentation on rehearsals

petr-muller opened the pull request:
/cc @stevekuznetsov @droslean @bbguimaraes

droslean commented on ci-operator/REHEARSALS.md:
```diff
@@ -0,0 +1,39 @@
+
+All pull requests trigger a `ci/prow/pj-rehearse` check, which detects the jobs
+affected by the proposed change. For each job selected for rehearsal (for
+various reasons, we are not able to rehearse all jobs), new Prowjob is
```
You already mention this parenthesis below

petr-muller replied:
That's intentional. Not everyone reads the whole thing. Here I mention it as a fact, and below I explain details about it.

droslean commented on ci-operator/REHEARSALS.md:
```diff
@@ -0,0 +1,39 @@
+rehearsing are logged in the output of the `ci/prow/pj-rehearse` job. Most
+importantly, we are currently *not* rehearsing template-based jobs.
+
+## Future work
```
Worth to mention, that in the future we will rehearse job not only for changed `spec`, but for other fields as well.

petr-muller replied:
addressed

stevekuznetsov commented on ci-operator/REHEARSALS.md:
```diff
@@ -0,0 +1,39 @@
+CI job rehearsals
+=================
+
+**This feature is not fully deployed to production yet**
```
This needs to be put into words a user will understand -- what is and is not deployed? What works? Maybe just specify what _is_ working

petr-muller replied:
addressed

petr-muller commented:
I have addressed the feedback in a new commit so it's easier to see how it was changed. I'd like to squash those commits before merge.

/hold
/retest

stevekuznetsov commented on ci-operator/REHEARSALS.md:
```diff
@@ -0,0 +1,55 @@
+
+At the moment, we are not able to rehearse all affected jobs for various
+reasons. Currently, we only rehearse `Presubmit` jobs affected by the
+following changes:
```
We must be diligent about updating this

droslean commented:
/lgtm

petr-muller commented:
/hold cancel
